File: problem_thirty.js
// perfect number 2
var value = 500
function perfectNumber(value){
    var arr =[]
    for(var i = 1 ; i< value ;i++){
        var count = 0
        for (var j = 1 ; j<i ; j++){
            if(i% j ==0){
                count += j
            }
        }
        if (count == i){
            arr.push(i)
        }

    }
    return arr
}
console.log(perfectNumber(value))

File: problem_thirty.py
# perfect number 2
value = int(input("Enter the number:"))
def MyFunction(value):
    lst =[]
    for i in range(1,value,1):
        count=0
        for j in range(1,i,1):
            if i % j == 0:
                count+=j
        if count == i:
            lst.append(i)
    return lst        

print(MyFunction(value))